File: app/src/main/java/com/parabola/newtone/mvp/view/fx/FxAudioSettingsView.java
package com.parabola.newtone.mvp.view.fx;

import moxy.MvpView;
import moxy.viewstate.strategy.AddToEndSingleStrategy;
import moxy.viewstate.strategy.OneExecutionStateStrategy;
import moxy.viewstate.strategy.StateStrategyType;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface FxAudioSettingsView extends MvpView {

    void setPlaybackSpeedSwitch(boolean enabled);
    void setPlaybackPitchSwitch(boolean enabled);


    @StateStrategyType(OneExecutionStateStrategy.class)
    void setPlaybackSpeedSeekbar(int progress);
    void setPlaybackSpeedText(float speed);

    @StateStrategyType(OneExecutionStateStrategy.class)
    void setPlaybackPitchSeekbar(int progress);
    void setPlaybackPitchText(float pitch);


    void hideBassBoostPanel();
    @StateStrategyType(OneExecutionStateStrategy.class)
    void setBassBoostSeekbar(int currentLevel);
    void setMaxBassBoostSeekbar(int maxStrength);
    void setBassBoostSwitch(boolean bassBoostEnabled);


    void hideVirtualizerPanel();
    @StateStrategyType(OneExecutionStateStrategy.class)
    void setVirtualizerSeekbar(int currentLevel);
    void setMaxVirtualizerSeekbar(int maxStrength);
    void setVirtualizerSwitch(boolean virtualizerEnabled);
}

File: app/src/main/java/com/parabola/newtone/mvp/view/fx/FxAudioSettingsView.kt
package com.parabola.newtone.mvp.view.fx;

import moxy.MvpView;
import moxy.viewstate.strategy.alias.AddToEndSingle;

@AddToEndSingle
public interface FxAudioSettingsView extends MvpView {

    void setPlaybackSpeedSwitch(boolean enabled);
    void setPlaybackSpeedSeekbar(int progress);
    void setPlaybackSpeedText(float speed);


    void setPlaybackPitchSwitch(boolean enabled);
    void setPlaybackPitchSeekbar(int progress);
    void setPlaybackPitchText(float pitch);


    void hideBassBoostPanel();
    void setBassBoostSeekbar(int currentLevel);
    void setBassBoostSwitch(boolean bassBoostEnabled);


    void hideVirtualizerPanel();
    void setVirtualizerSeekbar(int currentLevel);
    void setVirtualizerSwitch(boolean virtualizerEnabled);
}
